packages/villus/src/client.ts

```diff
@@ -17,14 +17,32 @@ import {
   QueryExecutionContext,
 } from './types';
 import { VILLUS_CLIENT } from './symbols';
-import { App } from 'vue';
+import { App, getCurrentInstance, inject } from 'vue';
 
 export interface ClientOptions {
   url: string;
   cachePolicy?: CachePolicy;
   use?: ClientPlugin[];
 }
 
+/**
+ * setActiveClient should be called to solve problem outside setup
+ */
+// eslint-disable-next-line no-use-before-define
+export let activeClient: Client | undefined;
+
+/**
+ * Sets or unsets the active client
+ *
+ * @param client - villus client instance
+ */
+export const setActiveClient = (client: Client | undefined) => (activeClient = client);
+
+/**
+ * Get the currently active client if there is any.
+ */
+export const getActiveClient = () => (getCurrentInstance() && inject(VILLUS_CLIENT)) || activeClient;
```

Comment:
I think we should leave out the `inject` logic from here and instead focus on the activeClient instance getter and setter in isolation. We can move the injection to the `injectWithSelf` call.

```suggestion
let activeClient: Client | undefined;

/**
 * Sets or unsets the active client
 *
 * @param client - villus client instance
 */
export const setActiveClient = (client: Client | undefined) => (activeClient = client);

/**
 * Get the currently active client if there is any.
 */
export const getActiveClient = () => activeClient;
```

Note that since we have a reliable getter now we can avoid exporting the active client instance.

Reply:
I worry about that if we getActiveClient, we can not assure injectWithSelf (resolveClient now) will be called first. so I write like below. 
```ts
export const getActiveClient = () => (getCurrentInstance() && inject(VILLUS_CLIENT)) || activeClient;
```
I also think if we can unite the inject to getActiveClient, just like:
```ts
export function getActiveClient(manualClient?: Client): Client {
  if (manualClient) {
  return manualClient
}
  const vm = getCurrentInstance() as any;
  let client = vm && inject(VILLUS_CLIENT);

  if (client) {
    setActiveClient(client);
} else {
   client = activeClient;   
}

  if (client === null || client === undefined) {
    throw new Error('Cannot detect villus Client, did you forget to call `useClient`? or you can pass a specific client to manualClient param. more details please read the documentation.');
  }

  return client;
}
```

what do you think? If you also think it is reasonable, I will change it and have a check where use injectWithSelf.

Reply:
By the way, I think it may be better to modify the error message, because I am not good at english, I just add some words after. you can make it, then I will add it to these code.

Reply:
> I worry about that if we getActiveClient, we can not assure resolveClient will be called first

Yes, that is a valid concern but `getActiveClient` is only used inside `resolveClient`. So there is no way it can get called before `resolveClient`.

> I also think if we can unite the inject to getActiveClient

Yep I see you did that in `resolveClient` now, it doesn't matter really. We can either say `getActiveClient` resolves the client or `resolveClient` is the one to do that. Not a big difference and I think the current implementation is fine 👍 

> By the way, I think it may be better to modify the error message

Sure, what do you think of this:

```
Cannot detect villus Client, did you forget to call `useClient`? Alternatively, you can explicitly pass a client as the `manualClient` argument.
```

Reply:
>  So there is no way it can get called before resolveClient. 
Yes, there is no problem now, just in case  someone will contribute to the code, they will not know the logic order at first (it is implicit). and worry we will forget it maybe extends code someday. 
> Not a big difference and I think the current implementation is fine
Agree that's fine now, so let's make it now
> Cannot detect villus Client, did you forget to call `useClient`? Alternatively, you can explicitly pass a client as the `manualClient` argument.
Clear and understandable. I will add it.